src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -194,12 +205,30 @@ class SDCPUThread : public ThreadContext
 
     /**
      * Entry point to begin work on a new instruction. Will also begin any
-     * related events, such as onBeginFetch to kick off any upcoming tasks
+     * related events, such as attemptFetch to kick off any upcoming tasks
      * necessary to complete that instruction. May issue an instruction
      * immediately if no new information is needed to set up a new instruction.
      */
     void advanceInst();
 
+    /**
+     * Entry point to fetch functionality. May be called more than once for a
+     * particular instruction, due to instructions spanning across fetch
+     * boundaries.
+     *
+     * @param inst A reference to the in-flight instruction object for which to
+     *  request fetch data.
+     */
+    void attemptFetch(std::weak_ptr<InflightInst> inst);
```

Comment:
Yeah, I think this should be combined with the decode logic.

Reply:
The other thing is that a decode could occur across the boundaries of two fetches, even with the buffer, so I think trying to combine it might require a more significant logic change.